docs/database-dynamodb-single-table.md

```diff
@@ -0,0 +1,92 @@
+# Database - DynamoDB Single Table
+
+The `database/dynamodb-single-table` construct deploys a single DynamoDB table with pre-configured indexes following [Single Table Design](https://www.alexdebrie.com/posts/dynamodb-single-table/) principles.
+
+## Quick start
+
+```yaml
+service: my-app
+provider:
+  name: aws
+
+constructs:
+    myTable:
+        type: database/dynamodb-single-table
+
+plugins:
+    - serverless-lift
+```
+
+On `serverless deploy`, a preconfigured DynamoDB table will be created.
+
+## How it works
+
+The `database/dynamodb-single-table` construct creates and configures the table for production:
+
+- a composite primary index with generic attributes names - `PK` for the partition key and `SK` for the sort key
+- 5 local secondary indexes, the maximum allowed, with generic names - `LSI-1` to `LSI-5` for the index names and `LSI-1-SK` to `LSI-5-SK` for the sort keys. Those indexes have no impact on pricing as long as their indexes are not populated with data. Modifying a table local secondary indexes configuration requires table re-creation, thus enabling all of them by default at table creation.
+- a configurable amount of up to 20 glocal secondary indexes with generic names - `GSI-1` to `GSI-20` for the index names, `GSI-1-PK` to `GSI-20-PK` for the partition keys and `GSI-1-SK` to `GSI-20-SK` for the sort keys
+- all indexes attributes have string data type, ideal for composite attribue - i.e. `value1#value2`
+- a DynamoDB stream publishing new and old values at each write operation on the table
```

Comment:
BTW do any of these settings incur additional costs? (for example the streams?)

Reply:
AFAIK, streams does not incur additional costs